pkg/resources/project_artefacts.go

```diff
@@ -0,0 +1,180 @@
+package resources
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/gthesheep/terraform-provider-dbt-cloud/pkg/dbt_cloud"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/customdiff"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var projectArtefactsSchema = map[string]*schema.Schema{
+	"project_id": &schema.Schema{
+		Type:        schema.TypeInt,
+		Required:    true,
+		Description: "Project ID",
+	},
+	"docs_job_id": &schema.Schema{
+		Type:        schema.TypeInt,
+		Optional:    true,
+		Description: "Docs Job ID",
+	},
+	"freshness_job_id": &schema.Schema{
+		Type:        schema.TypeInt,
+		Optional:    true,
+		Description: "Freshness Job ID",
+	},
+}
+
+func ResourceProjectArtefacts() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceProjectArtefactsCreate,
+		ReadContext:   resourceProjectArtefactsRead,
+		UpdateContext: resourceProjectArtefactsUpdate,
+		DeleteContext: resourceProjectArtefactsDelete,
+
+		CustomizeDiff: customdiff.All(
+			customdiff.ForceNewIfChange("project_id", func(_ context.Context, old, new, meta interface{}) bool { return true }),
+		),
+
+		Schema: projectArtefactsSchema,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+	}
+}
+
+func resourceProjectArtefactsCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	c := m.(*dbt_cloud.Client)
+
+	var diags diag.Diagnostics
+
+	projectID := d.Get("project_id").(int)
+	docsJobID := d.Get("docs_job_id").(int)
+	freshnessJobID := d.Get("freshness_job_id").(int)
+	projectIDString := strconv.Itoa(projectID)
+
+	project, err := c.GetProject(projectIDString)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if docsJobID != 0 {
+		project.DocsJobId = &docsJobID
+	} else {
+		project.DocsJobId = nil
+	}
```

Comment:
Currently handling `nil` that get automatically changed to `0` that way but it doesn't feel right...

Reply:
Think you may be able to deal with this by having them as optional/ pointers on the Project struct? then using omitempty in the json options?

Reply:
I have tried it but it didn't work. After some (way too much) digging, it looks like it is a limitation of of [SDKv2](https://stackoverflow.com/questions/73341494/is-it-possible-to-differentiate-between-a-zero-value-and-a-removed-property-in-t) that has been "fixed" in the new (but not stable yet) [Terraform Plugin Framewrok](https://developer.hashicorp.com/terraform/plugin/framework).

So despite the fact that it is not super clean I don't think that we can get around the small logic.

Reply:
@b-per - I've done the migration for the Plugin Framework on another provider, so will do it on this one at a later date and we can deal with it then